ding_unilogin/ding_unilogin.module

```diff
@@ -18,6 +18,21 @@ use Drupal\ding_unilogin\Exception\HttpException;
  * re-trigger it later (after logging in via UNI•Login).
  */
 function ding_unilogin_init() {
+  // Authenticated users with no "update" access to errors paths should be
+  // redirected to the error redirect path.
+  if (user_is_logged_in()) {
```

Comment:
I think there is an `ding_user_is_provider` (https://github.com/ding2/ding2/blob/cfea678b4dc7d5052247841f5a941e813a3fd2cc/modules/ding_user/ding_user.module#L1059) which should return `false` for all non-library users... could that be used for someting (not sure)?

Reply:
Good point, but we don't need to know if the user is authenticated through a provider. We just need to know if he's authenticated or not.